zio-http/src/main/scala/zhttp/http/Response.scala

```diff
@@ -75,12 +75,17 @@ final case class Response private (
 
     val jHeaders = self.headers.encode
     val jContent = self.data match {
-      case HttpData.Text(text, charset) => Unpooled.wrappedBuffer(text.getBytes(charset))
-      case HttpData.BinaryChunk(data)   => Unpooled.copiedBuffer(data.toArray)
-      case HttpData.BinaryByteBuf(data) => data
-      case HttpData.BinaryStream(_)     => null
-      case HttpData.Empty               => Unpooled.EMPTY_BUFFER
-      case HttpData.RandomAccessFile(_) => null
+      case HttpData.Asynchronous(_) => throw new IllegalStateException("Cannot encode response with incoming data")
```

Comment:
I don't think we need to throw here. We can simply ignore it.
Might need a unit test.